src/operators/OperatorState.h

```diff
@@ -42,9 +42,9 @@ namespace mrcpp {
 
 #define GET_OP_IDX(FT, GT, ID) (2 * ((GT >> ID) & 1) + ((FT >> ID) & 1))
 
-template <int D> class OperatorState final {
+template <int D, typename T> class OperatorState final {
```

Comment:
also here we can have a default for T, as the class is marked as `final` and we can't inherit from it anyway
```suggestion
template <int D, typename T = double> class OperatorState final {
```

Reply:
it is already set as default in mrcpp_declarations.h